### ISSUE TITLE
FasthttpProxyHTTPDialer() do not read env HTTP_PROXY?

### ISSUE THREAD

**Issue by Lallassu** [valyala/fasthttp#973]
When I use the `fasthttp/fasthttpproxy` `FasthttpProxyHTTPDialer()` it doesn't use the environment variable `HTTP_PROXY` as stated in the documentation: https://pkg.go.dev/github.com/valyala/fasthttp@v1.20.0/fasthttpproxy

I've initiated my client like this:
```
client := &fasthttp.Client{
		DialDualStack: true,
		Dial:          fasthttpproxy.FasthttpProxyHTTPDialer(),
}
```

If I print he `os.Getenv("HTTP_PROXY")`, right before I initiate the client, the environment variable is set but not used by fasthttp.

Side note: Using the `FasthttpHTTPDialer()` function and providing the proxy explicitly as argument works fine.

Am I doing anything wrong here?

**Comment by erikdubbelboer** [comment]
What does your value of `HTTP_PROXY` look like? We use https://pkg.go.dev/golang.org/x/net/http/httpproxy#FromEnvironment so it should follow the format stated there.

**Comment by Lallassu** [comment]
Yes, and I've tested the following:

`HTTP_PROXY=localhost:9999`
```
	proxier := httpproxy.FromEnvironment().ProxyFunc()
	x, err := proxier(&urlx.URL{Host: myurl, Scheme: "http"})
	fmt.Printf("%v, %v\n", x, err)
```
Right before I initiate the client. And this prints the correct proxy `http://localhost:9999`. But when doing the fasthttp client request, it's not used.

I've also tested to use a internal IP (192.168.1.x) towards localhost and still doesn't use it even though the proxy above shows correct proxy url.

I did see that the url and error from `proxier()` is nil/nil when running requests with localhost, but I've tested using 127.0.0.1 and the proxier returns the correct proxy in that case.
